Document unprivileged user in GitHub workflows

theory opened the pull request:
At some point GitHub changed things so that the entrypoint script does not run, so the `AS_USER` variable does nothing. Which is just as well, because that user would have no permissions, so even a checkout will fail.

So remove the GitHib-specific stuff from `entrypoint.sh` and separately document running the container with the CLI and as a GitHub workflow. Add a new section describing how to use `gosu` directly in a GitHub workflow to execute commands as an unprivileged user.

Thanks to @pgguru for figuring out the technique.

pgguru commented on README.md:
```diff
@@ -141,6 +129,31 @@ jobs:
 
+### Unprivileged User Workflow
+
+If you need to run take action as an unprivileged user, the `AS_USER` variable
+[documented above](#unprivileged-user) will not work, both because GitHub
+actions [require the root user] to work with the workspace and because GitHub
+does not execute `entrypoint` script.
+
+Instead, set things up as the root user, then use [gosu] to execute a command as
+the `postgres` (can still run `sudo`) or `nobody` (no privileges at all) user.
```
```suggestion
Note: GitHub Actions do not support the `AS_USER` facility when using this image
because they do not execute the `entrypoint` script.

If you _do_ need to take action as a lower-privileged user in a GitHub Action (such
as running the `initdb` command in a test script), you can use `gosu` in a `run:` block to execute the given task as either the `postgres`  or `nobody` user.
```

theory replied:
I think we were both editing this at the same time. The changes in 1b4556c fix the issue I think.

theory committed:
Document unprivileged user in GitHub workflows

At some point GitHub changed things so that the entrypoint script does
not run, so the `AS_USER` variable does nothing. Which is just as well,
because that user would have no permissions, so even a checkout will
fail.

So remove the GitHib-specific stuff from `entrypoint.sh` and separately
document running the container with the CLI and as a GitHub workflow.
Add a new section describing how to use `gosu` directly in a GitHub
workflow to execute commands as an unprivileged user.

Thanks to @pgguru for figuring out the technique.
